packages/ui-library/src/components/virtualized-list/index.tsx

```diff
@@ -1,37 +1,37 @@
 import React, { forwardRef } from 'react';
-import { Components, ItemContent, Virtuoso } from 'react-virtuoso';
+import { ItemContent, Virtuoso } from 'react-virtuoso';
 import styled from 'styled-components';
 
-const StyledList = styled.div`
+const StyledList = styled.div<{ $gap?: number }>`
   display: flex;
   flex-direction: column;
   flex: 1;
-  ${({ theme: { spacing } }) => `
-    gap: ${spacing(2)}
+  ${({ theme: { spacing }, $gap = 2 }) => `
+    gap: ${spacing($gap)}
   `}
 `;
 
-const VirtuosoComponents: Components = {
-  List: forwardRef<HTMLDivElement>(function VirtuosoList(props, ref) {
-    return <StyledList {...props} ref={ref} />;
-  }),
-};
 // eslint-disable-next-line @typescript-eslint/ban-types
 interface VirtualizedListProps<D, C = {}> {
   data?: readonly D[] | undefined;
   itemContent: ItemContent<D, C>;
   fetchMore?: () => void;
   context: C;
+  gap?: number;
 }
 
-const VirtualizedList = <D, C>({ data, itemContent, fetchMore, context }: VirtualizedListProps<D, C>) => (
+const VirtualizedList = <D, C>({ data, itemContent, fetchMore, context, gap }: VirtualizedListProps<D, C>) => (
   <Virtuoso
     style={{ height: '100%' }}
     data={data}
     itemContent={itemContent}
     endReached={fetchMore}
     context={context}
-    components={VirtuosoComponents}
+    components={{
+      List: forwardRef<HTMLDivElement>(function VirtuosoList(props, ref) {
```

Comment:
If we create it outside it doesnt get created on every re-render and saves up on memory